src/execute/engine.ts

```diff
@@ -29,9 +29,11 @@ import { mergeConfigs } from "../core/config.ts";
 import { ProjectContext } from "../project/types.ts";
 import { pandocBuiltInFormats } from "../core/pandoc/pandoc-formats.ts";
 import { gitignoreEntries } from "../project/project-gitignore.ts";
+import { juliaEngine } from "./julia.ts";
 
 const kEngines: ExecutionEngine[] = [
   knitrEngine,
+  juliaEngine,
   jupyterEngine,
   markdownEngine,
 ];
```

Comment:
One thing we'd like to do in the future is add the ability for third-party engines to register themselves as an extension, so that you don't have to ask us for a PR every time you want a new engine or change.

When you merge against `main`, you'll have to fix a small conflict here and it'll be clear what it's about. (There currently isn't the ability to actually register the engine externally, but I hope we'll do that in 2024 still.)